api/pkg/apis/v1alpha1/providers/stage/script/script.go

```diff
@@ -112,17 +112,29 @@ func downloadFile(scriptFolder string, script string, stagingFolder string) erro
 	}
 	tPath := filepath.Join(stagingFolder, script)
 
-	out, err := os.Create(tPath)
+	resp, err := http.Get(sPath)
 	if err != nil {
 		return err
 	}
-	defer out.Close()
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		return v1alpha2.NewCOAError(
+			nil,
+			"failed to download script because of "+string(body),
```

Comment:
Handle empty body?

Reply:
`body` will be the body of response, and it's not the reason of downloading failure. This field is for storing info. I changed the error string to make it more clear.